MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -1914,10 +1965,26 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
                 filters.Add("hwupload");
             }
 
-            if (state.DeInterlace("h264", true)
-                && string.Equals(outputVideoCodec, "h264_vaapi", StringComparison.OrdinalIgnoreCase))
+            var videoDecoder = GetHardwareAcceleratedVideoDecoder(state, options);
+
+            // If we are software decoding, and hardware encoding	  
+            if (string.Equals(outputVideoCodec, "h264_qsv", StringComparison.OrdinalIgnoreCase) 
+                && (string.IsNullOrEmpty(videoDecoder) || !videoDecoder.Contains("qsv")))
```

Comment:
```suggestion
                && (string.IsNullOrEmpty(videoDecoder) || !videoDecoder.Contains("qsv", StringComparison.OrdinalIgnoreCase)))
```
Maybe add braces to make the order more obvious.